common/encfstools.py

```diff
@@ -279,12 +283,13 @@ def __init__(self, cfg = None, profile_id = None, mode = None, parent = None,*ar
             self.mode = self.config.get_snapshots_mode(self.profile_id)
 
         self.parent = parent
+        self.read_only = read_only
         self.args = args
         self.kwargs = kwargs
 
-        self.ssh = sshtools.SSH(*self.args, symlink = False, **self.split_kwargs('ssh'))
-        self.rev_root = EncFS_mount(*self.args, symlink = False, **self.split_kwargs('encfs_reverse'))
-        super(EncFS_SSH, self).__init__(*self.args, **self.split_kwargs('encfs'))
+        self.ssh = sshtools.SSH(*self.args, symlink = False, read_only = self.read_only, **self.split_kwargs('ssh'))
+        self.rev_root = EncFS_mount(*self.args, symlink = False, read_only = self.read_only, **self.split_kwargs('encfs_reverse'))
```

Comment:
We don't need to mount ssh and encfs reverse with ro in here. These can be shared with other processes using rw. The top-layer encfs mount (line 292) mounting read-only is enough